src/com/spikes2212/genericsubsystems/drivetrains/SwerveDrivetrain.java

```diff
@@ -0,0 +1,67 @@
+package com.spikes2212.genericsubsystems.drivetrains;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.LimitedSubsystem;
+import com.spikes2212.genericsubsystems.commands.MoveLimitedSubsystemWithPID;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ *
+ */
+public abstract class SwerveDrivetrain extends Subsystem {
+	private static double tolerance = 1;
+	private SpeedController rotate1, rotate2, rotate3, rotate4, wheel1, wheel2, wheel3, wheel4;
```

Comment:
You don't bother yourself with implementation details, you just define a limited subsystem to rotate the wheels and let the implement it